simulator/func_sim/alu.h

```diff
@@ -291,6 +291,13 @@ struct ALU
         instr->v_dst[0] = ( instr->v_src[0] & bitmask<T>( pack_width)) | ( instr->v_src[1] << pack_width);
     }
 
+    template<typename I, typename T> static
+    void packu( I* instr)
+    {
+        auto pack_width = half_bitwidth<T>;
+        instr->v_dst[0] = ( instr->v_src[0] >> pack_width) | (( instr->v_src[1] >> pack_width) << pack_width));
```

Comment:
Can we use a bitmask instead of shifting left-right?

Reply:
I don't get how bitmask works

Reply:
See `pack` implementation and `bitmask<>()` unit tests:

https://github.com/MIPT-ILab/mipt-mips/blob/ebbaa5d2291ae6e2dc705ab0ec8124fdf0f72903/simulator/infra/t/unit_test.cpp#L94-L96

Reply:
I still dont get :(
It must look kinda?
 instr->v_dst[0] = ( instr->v_src[0] >> pack_width) | (bitmask<T>(pack_width) & instr->v_src[1])

Reply:
Yes. But I think you should double-check mask polarity.

Reply:
What does mean polarity?

Reply:
Polarity in schemotechnical meaning.
`bitmask(pack_width) & instr->v_src[1]` will take least significant bits from v_src[1]. Is it what you intend to do?

Reply:
No, i need upper. So it must be  instr->v_src[1] & bitmask(pack_width)

Reply:
Please double-check this assumption. Use your unit tests to verify your code.